rita_exit/src/rita_loop/mod.rs

```diff
@@ -65,9 +65,19 @@ pub fn start_rita_exit_loop() {
                 // setup exit clients and should crash if we fail to do so, otherwise we are preventing
                 // proper failover
                 let mut successful_setup: bool = false;
+                // cache of b19 routers we have successful rules and routes for
+                let mut wg_exit_clients: HashSet<WgKey> = HashSet::new();
+                // cache of b20 routers we have successful rules and routes for
+                let mut wg_exit_v2_clients: HashSet<WgKey> = HashSet::new();
 
                 loop {
-                    rita_exit_loop(&mut wg_clients, &mut debt_actions, &mut successful_setup)
+                    rita_exit_loop(
```

Comment:
in general you want to avoid mutable pointers are they are hard to reason about and debug. I'm sure you can refactor this such that it returns the data rather than taking a mutable reference.

Reply:
done